src/main/java/fi/aalto/cs/apluscourses/model/Exercise.java

```diff
@@ -163,7 +175,7 @@ public boolean isCompleted() {
   }
 
   public boolean isOptional() {
-    return maxSubmissions == 0 && maxPoints == 0;
+    return difficulty.equals("training") || difficulty.equals("challenge");
```

Comment:
Remark, for future reference: This is O1 specific and will be changed in #801.

Reply:
Or perhaps these values will stay forever and spread to all other courses as markings of tutorial exercises